field_audit/models.py

```diff
@@ -568,13 +573,42 @@ def delete(self, *, audit_action=AuditAction.RAISE):
         return value
 
     @validate_audit_action
-    def update(self, *args, audit_action=AuditAction.RAISE, **kw):
+    def update(self, audit_action=AuditAction.RAISE, **kw):
         if audit_action is AuditAction.IGNORE:
-            return super().update(*args, **kw)
-        else:
-            raise NotImplementedError(
-                "Change auditing is not implemented for update()."
+            return super().update(**kw)
+        assert audit_action is AuditAction.AUDIT, audit_action
+        from .field_audit import request
+        request = request.get()
+
+        fields_to_update = set(kw.keys())
+        audited_fields = set(
+            getattr(self.model, AuditEvent.ATTACH_FIELD_NAMES_AT)
+        )
+        fields_to_audit = fields_to_update & audited_fields
+        if not fields_to_audit:
+            # no audited fields are changing
+            return super().update(**kw)
+
+        values_to_fetch = list(fields_to_update) + ['pk']
```

Comment:
I think `pk` could already be in the set. A union would prevent it from being duplicated.

```suggestion
        values_to_fetch = fields_to_update | {"pk"}
```

Reply:
See 866af4c7bb877f02271a50cc55b9a7aadf9857c1